diff --git a/src/main.h b/src/main.h
index fa7acab..0cc7f20 100644
--- a/src/main.h
+++ b/src/main.h
@@ -127,8 +127,8 @@
 
 // DMRid database -----------------------------------------------
 
-#define DMRIDDB_USE_RLX_SERVER          1                                   // 1 = use http, 0 = use local file
-#define DMRIDDB_PATH                    "/xlxd/dmrid.dat"                   // local file path
+#define DMRIDDB_USE_RLX_SERVER          0                                   // 1 = use http, 0 = use local file
+#define DMRIDDB_PATH                    "/etc/mmdvm/xlxd/dmrid.dat"                   // local file path
 #define DMRIDDB_REFRESH_RATE            180                                 // in minutes
 
 
@@ -142,9 +142,9 @@
 // system paths -------------------------------------------------
 
 #define XML_PATH                        "/var/log/xlxd.xml"
-#define WHITELIST_PATH                  "/xlxd/xlxd.whitelist"
-#define BLACKLIST_PATH                  "/xlxd/xlxd.blacklist"
-#define INTERLINKLIST_PATH              "/xlxd/xlxd.interlink"
+#define WHITELIST_PATH                  "/etc/mmdvm/xlxd/xlxd.whitelist"
+#define BLACKLIST_PATH                  "/etc/mmdvm/xlxd/xlxd.blacklist"
+#define INTERLINKLIST_PATH              "/etc/mmdvm/xlxd/xlxd.interlink"
 #define DEBUGDUMP_PATH                  "/var/log/xlxd.debug"
 
 // system constants ---------------------------------------------
