--- a/Makefile
+++ b/Makefile
@@ -1,12 +1,13 @@
 all:	MMDVMCal
 
-LD  = g++
+CC  = gcc
 CXX = g++
 
 CXXFLAGS = -O2 -Wall
+LIBS = -lpthread
 
 MMDVMCal:	MMDVMCal.o SerialController.o SerialPort.o Console.o Utils.o
-		$(LD) $(LDFLAGS) -o MMDVMCal MMDVMCal.o SerialController.o SerialPort.o Console.o Utils.o $(LIBS)
+		$(CXX) $(LDFLAGS) -o MMDVMCal MMDVMCal.o SerialController.o SerialPort.o Console.o Utils.o $(LIBS)
 
 MMDVMCal.o:	MMDVMCal.cpp MMDVMCal.h SerialController.h Console.h Utils.h
 		$(CXX) $(CXXFLAGS) -c MMDVMCal.cpp
